### PR TITLE
finalize directory

### PR DESCRIPTION
**Pull request by shyaler**
While enumerating directory -> Adding the directory after adding all of its children +
markering the directory with folder end marker - to trigger finalize directory proccess.
In the Finalize proccess - add support to target compare mode.

please notice - this changes cause sync to work if nothing was changed in the source and the target.

### CODE REVIEW

File: cmd/syncComparator.go [hunk 1]
```diff
@@ -194,6 +194,8 @@ func (f *syncDestinationComparator) FinalizeTargetDirectory(relativeDir string,
 	f.sourceFolderIndex.lock.Unlock()
 	f.scannerLogger.Log(pipeline.LogInfo, fmt.Sprintf("Finalizing directory %s (FinalizeAll=%v)\n", relativeDir, finalizeAll))
 
+	isTargetCompare := f.cfdMode == common.CFDModeFlags.TargetCompare()
```
Comment (aymanjarrousms) [inline cmd/syncComparator.go:197]: our assumption that target compare is inly relevant to SMB, right?

**Review by aymanjarrousms — APPROVED** [review]

File: cmd/syncComparator.go (hunk 1 above)
Comment (shyaler) [reply cmd/syncComparator.go:197]: currently in storage mover scope yes

**Review by aymanjarrousms — DISMISSED** [review]

File: cmd/zc_traverser_file.go [hunk 2]
```diff
@@ -282,12 +295,13 @@ func newFileTraverser(rawURL *url.URL, p pipeline.Pipeline, ctx context.Context,
...
-	propertyGetter func(ctx context.Context) (azfilePropertiesAdapter, error)
-	entityType     common.EntityType
-	rootDirectory  bool
+	name                string
+	contentLength       int64
+	url                 url.URL
+	propertyGetter      func(ctx context.Context) (azfilePropertiesAdapter, error)
+	entityType          common.EntityType
+	rootDirectory       bool
+	isDirectoryEndMarke bool
```
Comment (shmuelberezin) [inline cmd/zc_traverser_file.go:304]: typo: isDirectoryEndMarke -> isDirectoryEndMarker

File: cmd/syncComparator.go [hunk 3]
```diff
@@ -307,13 +309,13 @@ func (f *syncDestinationComparator) FinalizeTargetDirectory(relativeDir string,
 	// last thing need to do the folder metaData updation incase of the finalizeAll true.
 	// TODO: We need to take care cfdMode == TargetCompare, as for it finalizer will be true. It cause each folder properties updation.
```
Comment (shmuelberezin) [inline cmd/syncComparator.go:310]: update this comment

**Review by shmuelberezin — CHANGES_REQUESTED** [review]

**Review by shmuelberezin — APPROVED** [review]

**Pull request merged**